clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -462,4 +474,19 @@ public void onComplete(ClientResponse response) {
             pendingCompletion.add(this);
         }
     }
+
+    /**
+     * When invoking poll from a multi-threaded environment, it is possible that the condition that
+     * the calling thread is awaiting has already been satisfied prior to the invocation of poll. We
+     * therefore introduce this interface to push the condition checking as close as possible to the
+     * invocation of poll.
```

Comment:
Is it worth mentioning that we actually perform the condition and poll while holding the same lock?